archetypes/Stake/FarmsTable/index.tsx

```diff
@@ -112,6 +113,16 @@ const PoolRow: React.FC<{
 
     const { bptDetails } = farm;
 
+    useEffect(() => {
+        if (
+            sessionStorage.getItem('portfolio.selectedToken') !== null &&
+            sessionStorage.getItem('portfolio.selectedToken') !== ''
+        ) {
+            onClickStake(TokenToFarmAddressMap(sessionStorage.getItem('portfolio.selectedToken')));
+            sessionStorage.setItem('portfolio.selectedToken', '');
+        }
+    }, []);
```

Comment:
Does this just open up the modal if there is a selected token?

Reply:
yep